cmd/kube-controller-manager/app/core.go

```diff
@@ -423,7 +423,7 @@ func startEphemeralVolumeController(ctx context.Context, controllerContext Contr
 	return nil, true, nil
 }
 
-const defaultResourceClaimControllerWorkers = 10
+const defaultResourceClaimControllerWorkers = 50
```

Comment:
Raising the default value might increase "CPU/Mem" utilization for all k8s consumers iiuc. 
Wondering what's the net increase that was seen in terms of cores/mem usage with this change ?

Reply:
I am not interested to know more on the too late for api change as we are not updating the api spec of the resource object right? why would adding a new flag to KCM providing an override be an issue? This change might work for 1000 node, but needs to be done for larger scale?

Reply:
>The CPU and mem increase will only be observed if all 50 go-routines are utilized right?

I am saying when all those 50 are utilized, it might lead to increase in CPU/Mem, I don't know by how much in this case, hence my earlier question. Many K8s customers would be running these components on specific boxes with certain CPU/Mem resources and might face unintended consequences if it leads to CPU throttling/OOM's on those boxes. To avoid those surprises, we generally avoid changing defaults IMO.

Reply:
> we are not updating the api spec of the resource object right

Command line and configuration options are also Kubernetes APIs which fall under API review and stability guidelines. IMHO if 50 is not enough, then raising it further through a command line option is not going to help with a more fundamental issue.

> I am saying when all those 50 are utilized, it might lead to increase in CPU/Mem

Each worker is doing some very light amount of work, so IMHO this shouldn't be an issue.

@alaypatel07 can correct me, but I think he had measured it.

Overall I am +1 for raising this default and then using the experience from more scale testing to decide about the need for a configuration option.